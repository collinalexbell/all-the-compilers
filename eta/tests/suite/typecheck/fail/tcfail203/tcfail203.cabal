name: tcfail203
version: 1.0
build-type: Simple
cabal-version: >= 1.2

library
    build-depends: base
    eta-options: -fdiagnostics-color=never
    exposed-modules: Tcfail203
                     Tcfail203a
