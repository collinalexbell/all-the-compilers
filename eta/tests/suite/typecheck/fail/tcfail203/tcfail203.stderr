
Tcfail203.hs:29:11: error:
    Pattern bindings containing unlifted types should use an outermost bang pattern:
      (I# x) = 5
    In an equation for ‘fail2’:
        fail2
          = 'a'
          where
              (I# x) = 5
   |
29 |     where (I# x) = 5
   |           ^^^^^^^^^^

Tcfail203.hs:32:11: error:
    Pattern bindings containing unlifted types should use an outermost bang pattern:
      (b, I# x) = (True, 5)
    In an equation for ‘fail3’:
        fail3
          = 'a'
          where
              (b, I# x) = (True, 5)
   |
32 |     where (b, I# x) = (True, 5)
   |           ^^^^^^^^^^^^^^^^^^^^^

Tcfail203.hs:41:11: error:
    Pattern bindings containing unlifted types should use an outermost bang pattern:
      (I# !x) = 5
    In an equation for ‘fail6’:
        fail6
          = 'a'
          where
              (I# !x) = 5
   |
41 |     where (I# !x) = 5
   |           ^^^^^^^^^^^

Tcfail203.hs:44:11: error:
    Pattern bindings containing unlifted types should use an outermost bang pattern:
      (b, !(I# x)) = (True, 5)
    In an equation for ‘fail7’:
        fail7
          = 'a'
          where
              (b, !(I# x)) = (True, 5)
   |
44 |     where (b, !(I# x)) = (True, 5)
   |           ^^^^^^^^^^^^^^^^^^^^^^^^
